### DISCUSSION TITLE
Darks in canon 450D

### DISCUSSION THREAD

**Discussion by jdemula** [aaronwmorris/indi-allsky#681] [Q&A] [answered]
Hello Aaron. It gives me an error in the generation of the dark master. It gets to save the darks in the temp folder, but it cuts off at the end of the process. I see some intermediate errors that I don't understand. I did several tests with different options but it gives the same error. I am attaching a txt with all the development that appears in the cli.
https://demulasky.com/indi-allsky/
[darks error.txt](https://github.com/aaronwmorris/indi-allsky/files/11173445/darks.error.txt)

**Comment by aaronwmorris** [comment]
I will get my DSLR back out in one or two days and test this.

It is possible your camera is not returning images in RAW format, but in JPEG format.  Make sure your camera is set to M mode and ensure you have selected RAW output in the settings.  I am not sure this is necessary, but I would try it.

**Reply by jdemula** [reply]
Thanks Aaron for your time. I always had it in M (BULB) but I checked and it was true that Mode (L) had it. Despite this, I was taking timelapse photos without a problem. I put it in RAW mode and the error it gives with the darks is still exactly the same. I'm insisting on doing it with the DSLR because I think it hurts me to leave these cameras in a drawer and not give them a chance. The quality seems very good to me and I saw an open window when I met indi-allsky.
![20230407_113512_HDR](https://user-images.githubusercontent.com/2183338/230593680-84d164c4-6120-434b-8bae-39618d558be2.jpg)
![20230407_113652_HDR](https://user-images.githubusercontent.com/2183338/230593687-cd5fb241-f31f-4b70-847c-3c045150dcb4.jpg)

**Reply by aaronwmorris** [reply]
I think it is great you are using a DSLR!

**Comment by aaronwmorris** [comment]
Your camera is indeed being set back to non-RAW mode.  Apparently, the default state of the indiserver is to set the image quality to `L` which is the first format (FORMAT_1) of the CCD_CAPTURE_FORMAT switch.  The RAW format is "FORMAT_9" on my camera.  Hopefully it is the same on others.

```
{
    "PROPERTIES" : {
        "CCD_INFO" : {
            "CCD_MAX_X" : 5184,
            "CCD_MAX_Y" : 3456,
            "CCD_PIXEL_SIZE" : 4.3,
            "CCD_PIXEL_SIZE_X" : 4.3,
            "CCD_PIXEL_SIZE_Y" : 4.3,
            "CCD_BITSPERPIXEL" : 16
        }
    },
    "SWITCHES": {
        "CCD_CAPTURE_FORMAT": {
            "on": [
                "FORMAT_9"
            ]
        }
    }
}
```

**Reply by jdemula** [reply, marked as answer]
Perfect Aaron. At first the format_9 did not work. I was testing in switches and format_7 is the correct one for the 450D.
And it already reads the temperature of the camera sensor! I did the darks with no problem and tonight it will be up and running again.

**Comment by czimm74** [comment]
As an FYI to this topic.  The Pi HQ is the same way, set to JPEG I could not generate darks.  But changed the output to RAW and it is working now.

**Reply by aaronwmorris** [reply]
I need to update the darks processing to ensure it verbosely fails when the camera does not send RAW data.  It is only possible to subtract darks with RAW data.

**Reply by czimm74** [reply]
I would also suggest adding that the configuration needs to be set to raw in the darks calibration wiki.  I wouldn't have figured it out if I didn't see this thread and make a guess I needed to do that.
